src/go/configinfo/service_info.go

```diff
@@ -408,35 +407,23 @@ func (s *ServiceInfo) processHttpRule() error {
 	return nil
 }
 
-func (s *ServiceInfo) addOptionMethod(originalMethod *methodInfo, path string) {
-	// All options have their operation as the following format: CORS_${suffix}.
-	// Appends ${suffix} to make sure it is not used by any http rules.
-	//
-	// b/145622434 changes ${suffix} to contain a url-safe path by replacing
-	// OpenAPI-specific characters from the UriTemplate. Spec here:
-	// https://swagger.io/docs/specification/paths-and-operations/
-	// This will ensure other services can correctly parse/display the operation name.
-	corsOperationBase := fmt.Sprintf("%s_CORS", util.AutogeneratedOperationPrefix)
-	formattedPath := strings.TrimPrefix(path, "/")
-	formattedPath = strings.ReplaceAll(formattedPath, "/", "_")
-	formattedPath = strings.ReplaceAll(formattedPath, "{", "")
-	formattedPath = strings.ReplaceAll(formattedPath, "}", "")
-	corsOperation := fmt.Sprintf("%s_%s", corsOperationBase, formattedPath)
-	genOperation := fmt.Sprintf("%s.%s", originalMethod.ApiName, corsOperation)
-
-	s.Methods[genOperation] = &methodInfo{
-		ShortName:  corsOperation,
-		ApiName:    originalMethod.ApiName,
-		ApiVersion: originalMethod.ApiVersion,
-		HttpRule: []*commonpb.Pattern{
-			{
-				UriTemplate: path,
-				HttpMethod:  util.OPTIONS,
-			},
-		},
-		IsGenerated: true,
-		BackendInfo: originalMethod.BackendInfo,
+func (s *ServiceInfo) addOptionMethod(originalMethod *methodInfo, path string) error {
+	genOperation := fmt.Sprintf("%s.%s_CORS_%s", originalMethod.ApiName, util.AutogeneratedOperationPrefix, originalMethod.ShortName)
```

Comment:
My concern is: this generated operation may not be unique since it only use apiName + ShortName
If it collapse with other cors method, if the other method has different backend,  they this CORS will route to a wrong backend.
I am not sure if ApiName + ShortName is unique enough
My understanding is ApiName +  ServiceName + Method,  are unique,  ShortName is Method. 
If there are two Rpc Services with same Method name under same Api,  we may have problem.

Reply:
+1. IMHO, the route won't be messed up since it is using regex path match -> cluster but the name may be duplicate.

Reply:
My concern is invalid.  Api is the same as gRPC service
ApiName + Method_name should be unique.

Reply:
Wayne and I discussed offline since I didn't fully understand, there should be no problems with that.

@TAOXUY can you clarify what problems you see? Keep in mind, `ApiName + . + MethodName` forms a selector, which is guaranteed to be unique. In fact, it's the regex path match that may not be unique, which is why PR #255 de-dups them. This PR still keeps that logic, as you can see by the new test case that was added.

Reply:
Same with Wayne concern, didn't see the ApiName included the serviceName. 

For the regex, I mean being unique after dedup in #255.